docs_website/docs/setup_guide/connect_to_query_engines.md

```diff
@@ -0,0 +1,296 @@
+---
+id: connect_to_query_engines
+title: Connect to Query Engines
+sidebar_label: Connect to Query Engines
+---
+
+## Overview
+
+Querybook supports all Sqlalchemy compatible query engines by default. Basic functionalities such as query execution, table metadata, and auto-completion are provided out of the box. However, more advanced integrations would require customized code. Overall, the query engines can be categorized into 3-tiers:
+
+| Tier                           | Tier 3     | Tier 2            | Tier 1             |
+| ------------------------------ | ---------- | ----------------- | ------------------ |
+| Summary                        | Not tested | Tested w/ DB      | Used in Production |
+| Library                        | Sqlalchemy | Custom/SqlAlchemy | Custom             |
+| Query Progress                 | x          | ?                 | ✓                  |
+| Query Logs                     | x          | ?                 | ✓                  |
+| Query Metadata                 | x          | ?                 | ✓                  |
+| Cancel Query                   | x          | ?                 | ✓                  |
+| User Authentication            | x          | x                 | ✓                  |
+| Syntax Error Parsing           | x          | ?                 | ✓                  |
+| Service discovery              | x          | x                 | ✓                  |
+| Language Specific Autocomplete | x          | x                 | ✓                  |
+
+Tier 1 engines do not mean they can be used in production everywhere since different companies/org require different kinds of integrations. However, tier 1 databases provide an excellent foundation to extend additional functionalities. Use them as a reference or subclass them via the [query engine plugin](../integrations/add_query_engine.md).
+
+If you have tried any of the tier 3 databases and confirmed it works, please update this doc to let others know.
+
+## Query Engine Support
+
+Querybook only supports a few of the Tier 2/1 databases by default. When Querybook is launched, it checks with SqlAlchemy to see if any of the databases below are available. If so, the query engine would be automatically available to set up in the Admin UI. Please see the [step by step guide](#step-by-step-guide) below to see an working example.
+
+## Step by step guide
+
+In this guide, we will through all the steps required to get Amazon Redshift to work in Querybook. Other query engines can be added similarly.
```

Comment:
In this guide, we will go through adding Amazon Redshift query engine to Querybook. This serves as an example to adding all query engines.